testing/test_metannotate_end_to_end.sh

```diff
@@ -14,38 +14,64 @@ python run_metannotate.py --orf_files=data/MetagenomeTest.fa --hmm_files=data/hm
 
 echo "Verifying outputs..."
 
-cd testing/test_constants
+EXIT_CODE=0
+ROOT_PATH=$(pwd)
 
-# Get reference hash file checksums
-FIRST_HASH=$(md5sum rpoB_0_msa_0.fa | cut -d ' ' -f 1)
-SECOND_HASH=$(md5sum rpoB_0_refseq_msa_1.fa | cut -d ' ' -f 1)
+## Check that the number of files in --output_dir is correct
+# TODO: replace this test with individual tests for each expected file
+NUM_OUTPUT_FILES=$(ls ${ROOT_PATH}/test_output | wc -l | sed 's/ //g')
+CORRECT_NUM_FILES=13
+if [[ ${NUM_OUTPUT_FILES} -eq ${CORRECT_NUM_FILES} ]]; then
+    echo "Correct number of files in output_dir ("${CORRECT_NUM_FILES}")"
+else
+    echo "Unexpected number of files in output_dir. Expected: "${CORRECT_NUM_FILES}"; Produced: "${NUM_OUTPUT_FILES}
+    EXIT_CODE=1
+fi
 
-cd ../..
+## Check hashes
+# Get reference hash file checksums
+declare -a REFERENCE_FILE_HASHES
+cd ${ROOT_PATH}/testing/test_constants
+REFERENCE_FILE_HASHES[0]=$(md5sum rpoB_0_msa_0.fa | cut -d ' ' -f 1)
+REFERENCE_FILE_HASHES[1]=$(md5sum rpoB_0_refseq_msa_1.fa | cut -d ' ' -f 1)
+# only first 14 columns of annotations table are consistent between runs
+REFERENCE_FILE_HASHES[2]=$(cut -f 1-14 rpoB_0_MetagenomeTest_0_annotations.tsv | md5sum | cut -d ' ' -f 1)
 
 # Store generated test output hashes in array
+# Keys in REFERENCE_FILE_HASHES should correspond to same key in REGENERATED_FILE_HASHES
 declare -a REGENERATED_FILE_HASHES
-
-for entry in $(ls test_output); do
-    # Checks if entry is the one we are looking to compare
-    if [[ ${entry} == *"0_msa"* && ${entry} == *".fa"* ]]; then
-        cd test_output
+declare -a CHECKED_FILES
+cd ${ROOT_PATH}/test_output
+for entry in $(ls); do
+    # Checks if entry is the query msa fasta
+    if [[ ${entry} == rpoB_0_msa*fa ]]; then
         REGENERATED_FILE_HASHES[0]=$(md5sum ${entry} | cut -d ' ' -f 1)
-        cd ..
+        CHECKED_FILES[0]=${entry}
     fi
 
-    # Checks if entry is the one we are looking to compare
-    if [[ ${entry} == *"refseq"* && ${entry} == *".fa"* ]]; then
-        cd test_output
+    # Checks if entry is the refseq msa fasta
+    if [[ ${entry} == rpoB_0_refseq_msa*fa ]]; then
```

Comment:
The next two if's should be ```else if``` (https://ryanstutorials.net/bash-scripting-tutorial/bash-if-statements.php.). 
We know that if the file enters is ```rpoB_0_msa*fa``` then it can't be ```rpoB_0_refseq_msa*fa``` or ```rpoB_0_MetagenomeTest_0_annotations*tsv``` so we can not do those if statements and go to the next iteration of the loop.